_posts/2018-04-28-hackillinois-2018-recap.markdown

```diff
@@ -1,28 +1,47 @@
+Attendees worked late into the night starting on their projects: mentors taught students how to set up their environments and basics of the project and how to make contributions.
 
-Still, it wasn't all about pure development. Sppnsors provided plenty of fun activities to take a break, including video games and puzzles in their lounges, for instance. Many attendees also stepped out for fun games of nerf or to play with the therapy dogs.
+![food](/assets/article_images/food.jpg "Attendees fill up on food to fuel them through the night")
 
-As the hacking deadline approached Sunday morning, groups worked hard to finish up and put on the final touches. Groups presented their open-source contributions to our judges and everyone else in attendence.
+HackIllinois also featured a wide array of talks and workshops. There were many presentations on interesting techical topics such as machine learning and virtual reality, as well as sessions focused specifically on open source, such as the Women in Open Source panel. Some attendees also prepared short lightning talks on a variety of diverse topics, from networking to memes.
 
-After the closing ceremony, students piled onto buses back to Chicago, Indiana, Michigan, and even Georgia and Florida! And with that, HackIllinois 2018 came to a close.
+![women in open source panel](/assets/article_images/wos.jpg "Lively discussion at the women in open source panel")
 
-Watch out for registration for HackIllinois 2019 on our website at [hackillinois.org](https://hackillinois.org)! And if you're intested in getting involved as a sponsor or mentor, be sure to reach out to us at [contact@hackillinois.org].
+There were also some great keynotes from members of the open source community. Travis Oliphant, the creator of NumPy and SciPy and the CEO of Anaconda, spoke about open source and changes in data science. Vijay Patel, a software engineer at Google, discussed his work on gRPC and other open source projects in his time at Google. Katy Huff, an associate professor at Illinois, and Steve Klabnik, a member of the Rust core team, both also spoke about their experiences in open source.
+
+![Travis Oliphant keynote](/assets/article_images/travis.jpg "Travis Oliphant talks about his company Quansight in his keynote")
+
+Throughout the weekend, there were all sorts of activities to take a break from coding. Our sponsors set out swag at their tables and games in their lounges.
+
+![Pickles the wolf](/assets/article_images/pickles.jpg "Pickles the wolf hangs out in the IMO lounge")
+
+Tons of attendees had fun playing Nerf at the Kenney Gym on Saturday. As is tradition, the last game was a Hackillinois staff versus sponsor round.
+
+![Nerf battle](/assets/article_images/nerf.jpg "Battling it out at Nerf")
 
+Sunday morning brought the end of hacking, and groups wrapped up their contributions and documented their work on Devpost in preparation for demos. Some therapy dogs visited and brought a smile to many a face.
```

Comment:
is "brought a smile to many a face" a phrase? would probably make more sense with "brought smiles to many faces"